lib/sorbet-rails/model_plugins/active_record_assoc.rb

```diff
@@ -131,19 +131,12 @@ def populate_collection_assoc_getter_setter(assoc_module_rbi, assoc_name, reflec
 
   sig { params(reflection: T.untyped).returns(T.nilable(T::Boolean)) }
   def assoc_should_be_untyped?(reflection)
-    polymorphic_assoc?(reflection) || !@available_classes.include?(reflection.klass.name)
+    reflection.polymorphic? || !@available_classes.include?(reflection.klass.name)
   end
 
   sig { params(reflection: T.untyped).returns(T.nilable(T::Boolean)) }
   def relation_should_be_untyped?(reflection)
     # only type the relation we'll generate
     assoc_should_be_untyped?(reflection) || !@available_classes.include?(reflection.klass.name)
   end
-
-  sig { params(reflection: T.untyped).returns(T.nilable(T::Boolean)) }
-  def polymorphic_assoc?(reflection)
-    reflection.through_reflection ?
-      polymorphic_assoc?(reflection.source_reflection) :
-      reflection.polymorphic?
```

Comment:
This recursive logic is to determine the polymorphic-ness of all assoc in the chain. I think it is still valid and we should keep it. I definitely remember running into a case that needed it and had to implement the recursive check.

Reply:
Interesting.

When I was implementing this functionality in our model RBI rake task the issue that prompted me to handle the polymorphic case is that `reflection.klass` blows up if the `klass` can't be determined. I'm assuming that's why we're trying to be more thorough here to determine if this can be polymorphic.

I actually wound up taking a more empirical approach any just catching `reflection.klass` blowing up and setting the assoc_class to `T.untyped` in that case. I've updated the PR to use the approach more similar to the one we used in our rake task.